statsmodels/base/distributed_estimation.py

```diff
@@ -0,0 +1,436 @@
+
+    Parameters
+    ----------
+    mod : statsmodels model class
+        The model for the current machine.
+    params : array-like
+        The estimated coefficients for the current machine.
+    hess_kwds : dict-like or None
+        Keyword arguments for the hessian function.
+
+    Returns
+    -------
+    An array-like object, updated design matrix, same dimension
+    as mod.exog
+    """
+
+    # TODO need to handle duration and other linear model classes
+    rhess = np.sqrt(mod.hessian_obs(np.array(params), **hess_kwds))
+    return rhess[:, None] * mod.exog
+
+
+def _gen_gamma_hat(X_beta, pi, alpha):
+    """generates the gamma hat values for the pith variable, used to
+    estimate theta hat.
+
+    Parameters
+    ----------
+    X_beta : array-like
+        The weighted design matrix for the current machine
+    pi : scalar
+        Index of the current variable
+    alpha : scalar or array-like
+        The penalty weight.  If a scalar, the same penalty weight
+        applies to all variables in the model.  If a vector, it
+        must have the same length as `params`, and contains a
+        penalty weight for each coefficient.
+
+    Returns
+    -------
+    An array-like object of length p-1
+
+    Notes
+    -----
+
+    gamma_hat_i = arg min 1/(2n) ||X_beta,i - X_beta,-i gamma||_2^2
+                          + alpha ||gamma||_1
+    """
+
+    p = X_beta.shape[1]
+    # handle array alphas
+    if not np.isscalar(alpha):
+        alpha = alpha[ind]
+
+    ind = list(range(p))
+    ind.pop(pi)
+
+    # TODO use elastic net optimization routine directly
+    tmod = OLS(X_beta[:, pi], X_beta[:, ind])
+
+    gamma_hat = tmod.fit_regularized(alpha=alpha).params
+
+    return gamma_hat
+
+
+def _gen_tau_hat(X_beta, gamma_hat, pi, alpha):
+    """generates the tau hat value for the pith variable, used to
+    estimate theta hat.
+
+    Parameters
+    ----------
+    X_beta : array-like
+        The weighted design matrix for the current machine
+    gamma_hat : array-like
+        The gamma_hat values for the current variable
+    pi : scalar
+        Index of the current variable
+    alpha : scalar or array-like
+        The penalty weight.  If a scalar, the same penalty weight
+        applies to all variables in the model.  If a vector, it
+        must have the same length as `params`, and contains a
+        penalty weight for each coefficient.
+
+    Returns
+    -------
+    A scalar
+
+    Notes
+    -----
+
+    tau_hat_i = sqrt(1/n ||X_beta,i - X_beta,-i gamma_hat||_2^2
+                     + alpha ||gamma||_1)
+    """
+
+    n, p = X_beta.shape
+    # handle array alphas
+    if not np.isscalar(alpha):
+        alpha = alpha[ind]
+
+    ind = list(range(p))
+    ind.pop(pi)
+
+    d = np.linalg.norm(X_beta[:, pi] - X_beta[:, ind].dot(gamma_hat))**2
+    d = np.sqrt(d / n + alpha * np.linalg.norm(gamma_hat, 1))
+    return d
+
+
+def _gen_theta_hat(gamma_hat_l, tau_hat_l, p):
```

Comment:
I think in most cases where you have a function name including "gen" it should be changed to "get" or "calc".  "gen" implies a Python generator to me, which these are not.